rest_tools/server/handler.py

```diff
@@ -102,22 +119,28 @@ def get_template_namespace(self):
         return namespace
 
     def get_current_user(self):
+        """Get the current user, and set auth-related attributes."""
         try:
-            type,token = self.request.headers['Authorization'].split(' ', 1)
+            type, token = self.request.headers['Authorization'].split(' ', 1)
             if type.lower() != 'bearer':
                 raise Exception('bad header type')
             logger.debug('token: %r', token)
             data = self.auth.validate(token)
             self.auth_data = data
             self.auth_key = token
+            wtt.get_current_span().set_attribute('self.auth_data.role', self.auth_data['role'])
```

Comment:
I think this line is what has been setting off my unit tests in LTA.

This is the data payload from a real token that I generated for using LTA:
```
{
  "aud": "ANY",
  "ver": "scitoken:2.0",
  "name": "Patrick Meade",
  "scope": "lta:admin file-catalog:admin",
  "iss": "https://tokens.icecube.wisc.edu",
  "sub": "pmeade",
  "exp": 1638893413.372062,
  "nbf": 1607357413.372062,
  "iat": 1607357413.372062,
  "type": "service"
}
```

I think the lack of a `role` field in tokens causes a `KeyError: 'role'` exception to be generated here. When it falls into the `except Exception` block below, it looks like a `failed auth`. Adding a guard (`if "role" in`...) that lets me bypass the line also fixes my unit tests.

@ric-evans 
Do we want `sub` here, to know who?
Or are we looking to parse `scope` here, to see what the role the token is granting in the application?

I'll create another review with a deeper dive into everything else; but I hope this review is enough to get started on a fix?

Reply:
That would make sense. `scope` looks like the best parallel. `sub` would also be helpful. Whatever you think is best. I'll review a PR if you put one in :+1:

Reply:
@blinkdog or just make a push, that's cool too! :smile:

Reply:
Yeah, a guard is probably the thing to do, so if it doesn't exist it doesn't get added to the span.